aspnetcore/blazor/tutorials/movie-database-app/part-3.md

```diff
@@ -0,0 +1,613 @@
+                   for .NET 10 work on handling a 404 with static SSR without \
+                   having to navigate to a non-existent endpoint. -->
+
+There isn't an actual `notfound` endpoint (Razor component) in the app. When adopting server-side rendering (SSR), Blazor doesn't have a mechanism to return a 404 (Not Found) status code. As a temporary workaround, a 404 is generated by navigating to a non-existent endpoint. This scaffolded code is for your further implementation of a suitable result when not finding an entity. For example, you could have the component direct the user to a page where they can file an inquiry with your support team, or you could remove the injected <xref:Microsoft.AspNetCore.Components.NavigationManager> and <xref:Microsoft.AspNetCore.Components.NavigationManager.NavigateTo%2A?displayProperty=nameWithType> code and replace it with Razor markup and code that displays a message to the user that the entity wasn't found.
+
+### `Create` component
+
+Open the `Create` component definition file (`Components/Pages/Movies/Create.razor`).
+
+The component uses a built-in component called an <xref:Microsoft.AspNetCore.Components.Forms.EditForm>, which renders a form for user input and includes validation features.
+
+CSS classes aren't present in the following example to simplify the display:
+
+```razor
+<EditForm method="post" Model="Movie" OnValidSubmit="AddMovie" FormName="create" Enhance>
+    <DataAnnotationsValidator />
+    <ValidationSummary />
+    <div>
+        <label for="title">Title:</label> 
+        <InputText id="title" @bind-Value="Movie.Title" /> 
+        <ValidationMessage For="() => Movie.Title" /> 
+    </div>
+    <div>
+        <label for="releasedate">ReleaseDate:</label> 
+        <InputDate id="releasedate" @bind-Value="Movie.ReleaseDate" /> 
+        <ValidationMessage For="() => Movie.ReleaseDate" /> 
+    </div>
+    <div>
+        <label for="genre">Genre:</label> 
+        <InputText id="genre" @bind-Value="Movie.Genre" /> 
+        <ValidationMessage For="() => Movie.Genre" /> 
+    </div>
+    <div>
+        <label for="price">Price:</label> 
+        <InputNumber id="price" @bind-Value="Movie.Price" /> 
+        <ValidationMessage For="() => Movie.Price" /> 
+    </div>
+    <button type="submit">Create</button>
+</EditForm>
+```
+
+The <xref:Microsoft.AspNetCore.Components.Forms.EditForm.Model%2A> parameter is assigned the model, in this case `Movie`. <xref:Microsoft.AspNetCore.Components.Forms.EditForm.OnValidSubmit%2A> specifies a method to invoke (`AddMovie`) when the form is submitted. By convention, every form should assign a <xref:Microsoft.AspNetCore.Components.Forms.EditForm.FormName%2A> to prevent form collisions when multiple forms are present on a page. The <xref:Microsoft.AspNetCore.Components.Forms.EditForm.Enhance%2A> flag activates a Blazor feature for server-side rendering (SSR) that submits the form without performing a full-page reload.
```

Comment:
I didn't state that because client-side validation doesn't work with static SSR. Because it occurs on the server, it seemed misleading to imply the checks are made client-side when the form is submitted.

Reply:
Without this clarification, the text suggests that OnValidSubmit will always be called when the form is submitted, which isn't the case. The data is still validated on the server and it must be valid for the OnValidSumit method to be called. I don't see why it matters whether this validation was done on the client or the server.

Reply:
I only mean that a dev might mistakenly think that the form doesn't POST with invalid data. I think an added remark that validation with static SSR only occurs on the server would cover the idea.